src/vm/amd64/PInvokeStubs.asm

```diff
@@ -13,6 +13,23 @@ include AsmConstants.inc
 
 extern GenericPInvokeCalliStubWorker:proc
 extern VarargPInvokeStubWorker:proc
+extern JIT_RareDisableHelper:proc
+
+extern s_gsCookie:QWORD
+extern ??_7InlinedCallFrame@@6B@:QWORD
+extern g_TrapReturningThreads:DWORD
+
+; Min amount of stack space that a nested function should allocate.
+MIN_SIZE equ 28h
+
+REMOVE_FRAME_FROM_THREAD macro frameReg, threadReg, trashReg
```

Comment:
A nit - could you please call the macro POP_FRAME_FROM_THREAD instead to match how we call the operation in  c++ code?

Reply:
I'll rename it.

Reply:
Another option is to move the popping of the frame on the slow path into the C helper. If you do that the need for this macro will disapper and you will have bit less of assembly code to maintain which is always goodness.

Reply:
R2RDump sources are under https://github.com/dotnet/coreclr/tree/master/src/tools/r2rdump . We typically update the dumper together with the file format change.